src/main/java/com/nike/cerberus/event/processor/AuditLogProcessor.java

```diff
@@ -0,0 +1,94 @@
+
+/**
+ * Event Processor that only cares about auditable events and outputs to a special audit log appender in a flat json
+ * format that is optimized for use with AWS Athena
+ */
+public class AuditLogProcessor implements EventProcessor {
+
+    protected final Logger auditLogger = LoggerFactory.getLogger(this.getClass());
+
+    private static final DateTimeFormatter ATHENA_DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm:ss");
+    private static final String PARTY_LIKE_ITS_99 = "1999-01-01T01:00:00+00:00";
+
+    private final ObjectMapper om = new ObjectMapper();
+
+    @Override
+    public void process(Event originalEvent) {
+        getAuditableEvent(originalEvent).ifPresent(event -> {
+
+            Optional<CerberusPrincipal> cerberusPrincipal = event.getPrincipalAsCerberusPrincipal();
+
+            ImmutableMap<String, String> flattenedAuditEvent = ImmutableMap.<String, String>builder()
+                    .put("event-timestamp", event.getTimestamp().format(ATHENA_DATE_FORMATTER))
+                    .put("principal-name", event.getPrincipalName())
+                    .put("principal-type", cerberusPrincipal
+                            .map(p -> cerberusPrincipal.get().getPrincipalType().getName()).orElse("Unknown"))
+                    .put("principal-token-created", cerberusPrincipal
+                            .map(p -> cerberusPrincipal.get().getTokenCreated().format(ATHENA_DATE_FORMATTER))
+                            .orElseGet(() -> OffsetDateTime.parse(PARTY_LIKE_ITS_99, ISO_OFFSET_DATE_TIME).format(ATHENA_DATE_FORMATTER)))
+                    .put("principal-token-expires", cerberusPrincipal
+                            .map(p -> cerberusPrincipal.get().getTokenExpires().format(ATHENA_DATE_FORMATTER))
+                            .orElseGet(() -> OffsetDateTime.parse(PARTY_LIKE_ITS_99, ISO_OFFSET_DATE_TIME).format(ATHENA_DATE_FORMATTER)))
+                    .put("principal-is-admin", cerberusPrincipal
+                            .map(p -> String.valueOf(p.isAdmin())).orElseGet(() -> String.valueOf(false)))
+                    .put("ip-address", event.getIpAddress())
+                    .put("x-forwarded-for", event.getxForwardedFor())
+                    .put("client-version", event.getClientVersion())
+                    .put("http-method", event.getMethodAsString())
+                    .put("path", event.getPath())
+                    .put("action", event.getAction())
+                    .put("name", event.getName())
+                    .put("sdb-name-slug", event.getSdbNameSlug())
+                    .put("originating-class", event.getOriginatingClass())
+                    .build();
+
+            try {
+                auditLogger.info(om.writeValueAsString(flattenedAuditEvent));
+            } catch (JsonProcessingException e) {
+                LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME).error("failed to log audit event", e);
```

Comment:
Is that really the logger you want?

Reply:
Yes, want the errors going to the main log appender and not the audit log

Reply:
I mean "ROOT" is fine with me, but It can be any string other than this.getClass()